spec/emerald/parser_spec.rb

```diff
@@ -164,6 +164,23 @@
                                            Emerald::List.new( Emerald::Atom.new('+'), Emerald::Number.new(3),
                                                             Emerald::List.new(  Emerald::Atom.new('-'), Emerald::Number.new(4) ) ) ) ] )
     end
+    
+    it "parses a list which contains a list and a number to the right of the internal list" do
+      source = '(+ (- 0 1) 2))'
+      ast = Emerald::Parser.new(source).parse
+      expect(ast).to eq([Emerald::List.new( Emerald::Atom.new('+'),  
+                                           Emerald::List.new( Emerald::Atom.new('-'), Emerald::Number.new(0), Emerald::Number.new(1) ),
+                                                             Emerald::Number.new(2) ) ] )
+    end
+
+    it "parses nested lists with numbers to the left of the lists" do
+      source = '(+ 2 (+ 3 (+ 0) 4) 3 4)'
+      ast = Emerald::Parser.new(source).parse
+      expect(ast).to eq([Emerald::List.new( Emerald::Atom.new('+'), Emerald::Number.new(2), 
+                                           Emerald::List.new( Emerald::Atom.new('+'), Emerald::Number.new(3) , 
+                                                             Emerald::List.new( Emerald::Atom.new('+'), Emerald::Number.new(0) ),
+                                                             Emerald::Number.new(4) ), Emerald::Number.new(3), Emerald::Number.new(4) ) ] )
```

Comment:
Yep, I agree they're getting a little ridiculous... But I wanted to check this kind of scenario. Yep, could add a linter. Do you think a shorter test would be sufficient? E.g.:
```
source = '(+ 2 ) 3 4 5)'
```
or even
```
source = '(+ 2 ) 3 4 )'
```